src/celeritas/optical/ScintillationParams.cc

```diff
@@ -0,0 +1,32 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/optical/ScintillationParams.cc
+//---------------------------------------------------------------------------//
+#include "ScintillationParams.hh"
+
+#include "corecel/cont/Range.hh"
+#include "corecel/data/CollectionBuilder.hh"
+#include "celeritas/Types.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Construct with optical property data.
+ */
+ScintillationParams::ScintillationParams(OpticalPropertyCRef const& properties)
+{
+    HostVal<ScintillationData> data;
+    for (auto mid : range(OpticalMaterialId(properties.scint_spectra.size())))
+    {
+        make_builder(&data.spectra).push_back(properties.scint_spectra[mid]);
```

Comment:
```suggestion
    CollectionBuilder spectra{&data.spectra};
    for (auto mid : range(OpticalMaterialId(properties.scint_spectra.size())))
    {
        spectra.push_back(properties.scint_spectra[mid]);
```

Reply:
Or in this case I think just `data.spectra = properties.scint_spectra` would work (although again, not sure why they are stored twice).

Reply:
Build it directly from ScintillationData.